cmd/agent/app/dogstatsd_stats.go

```diff
@@ -94,14 +96,23 @@ func requestDogstatsdStats() error {
 		}
 	}
 
-	if dsdStatsFilePath != "" {
-		if err := ioutil.WriteFile(dsdStatsFilePath, []byte(s), 0644); err != nil {
-			fmt.Println("Error while writing the output:", err)
-		} else {
-			fmt.Println("Dogstatsd stats written in:", dsdStatsFilePath)
+	if dsdStatsFilePath == "" {
+		fmt.Println(s)
+		return nil
+	}
+
+	// if the file is already existing, ask for a confirmation.
+	if _, err := os.Stat(dsdStatsFilePath); err == nil {
```

Comment:
I should know this, but I can't remember for sure. How does this behave with symbolic links? Say `dsdStatsFilePath` is a symlink. When you write to it, will you be overriding the symlink into a file or will you be overwriting the symlink target. I think it might be the latter, and we should maybe handle that case differently or bail out.

Reply:
The symlink is correctly followed and the content is correctly written in the target, not overriding the symlink 👍